datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -1754,6 +1754,76 @@ def create_new_revision_from_existing(  # pylint: disable=too-many-locals,too-ma
     return new_revision
 
 
+@router.post(
+    "/nodes/{name}/refresh/",
+    response_model=NodeOutput,
+    status_code=201,
+)
+def refresh_source_node(
+    name: str,
+    *,
+    session: Session = Depends(get_session),
+    query_service_client: QueryServiceClient = Depends(get_query_service_client),
+):
+    """
+    Refresh a source node with the latest columns from the query service.
+    """
+    source_node = get_node_by_name(session, name, node_type=NodeType.SOURCE)
+    current_revision = source_node.current
+
+    # Get the latest columns for the source node's table from the query service
+    columns = query_service_client.get_columns_for_table(
+        current_revision.catalog.name,
+        current_revision.schema_,  # type: ignore
+        current_revision.table,  # type: ignore
+        current_revision.catalog.engines[0]
+        if len(current_revision.catalog.engines) >= 1
+        else None,
+    )
+
+    # Check if any of the columns have changed (only continue with update if they have)
+    column_changes = {col.identifier() for col in current_revision.columns} != {
+        col.identifier() for col in columns
+    }
+    if not column_changes:
+        return source_node
+
+    # Create a new node revision with the updated columns and bump the version
+    old_version = Version.parse(source_node.current_version)
+    new_revision = NodeRevision.parse_obj(current_revision.dict(exclude={"id"}))
+    new_revision.version = str(old_version.next_major_version())
+    new_revision.columns = [
+        Column(name=column.name, type=column.type, node_revisions=[new_revision])
+        for column in columns
+    ]
+
+    # Keep the dimension links and attributes on the columns from the node's
+    # last revision if any existed
+    new_revision.copy_dimension_links_from_revision(current_revision)
+
+    # Point the source node to the new revision
+    source_node.current_version = new_revision.version
```

Comment:
I wonder when we'll get the first user request like: "hey, is it possible to like, set my source node to temporarily not be validated by DJ and just stay as is until I finish some work on my table." But I am sure if we really need to worry about delivering this feature right away.

Reply:
Good point - maybe we can add a `refresh: bool` setting to source nodes that defaults to true but can be set to false by the source node's owner. That way they can control if they want the node to be actioned on by the reflection service.